### DISCUSSION TITLE
LTree

### DISCUSSION THREAD

**Discussion by cemoktra** [SeaQL/sea-orm#197] [Q&A]
Is Postgres LTree supported? I dont see any occurences in code or documentation?

**Comment by billy1624** [comment]
What is the corresponding Rust type of `LTree`? A String like path?

**Reply by cemoktra** [reply]
Diesel apparently uses a `String`

**Reply by billy1624** [reply]
We depends on db driver `sqlx` to decode and encode various db types and Rust type. So, it comes down to weather `sqlx` supports representing `LTree` Postgres type as `String` Rust type or not. We haven't test it in our test case. We only tested basic Postgres types.

**Reply by billy1624** [reply]
You may test it now by changing the Model attribute of the LTree column to String type

**Reply by holmofy** [reply]
@billy1624  Can it be used like this?

```rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "paper_question")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub paper_id: i32,
    #[sea_orm(primary_key, auto_increment = false)]
    pub question_id: i32,
    pub sort: i16,
    #[sea_orm(column_type = "LTree")]
    pub keypoint_path: String,
    #[sea_orm(column_type = "Float")]
    pub correct_ratio: f32,
}
```

But it reported an error: 
```sh
Caused by:
    0: Query Error: error returned from database: column "keypoint_path" is of type ltree but expression is of type text
    1: error returned from database: column "keypoint_path" is of type ltree but expression is of type text
    2: error returned from database: column "keypoint_path" is of type ltree but expression is of type text
    3: column "keypoint_path" is of type ltree but expression is of type text
```

**Reply by Tim-Otte** [reply]
Is there an update regarding this topic? Apparently SeaQL already supports LTree as a column type and also supports migrating colmns with the ltree type, but I can't find any documentation regarding inserting/updating entities.

I'm getting the same error as @holmofy and the only solution I currently see is that I have to manually write the queries to insert and update the entities with an ltree column.

Thanks already for your help :heart:
